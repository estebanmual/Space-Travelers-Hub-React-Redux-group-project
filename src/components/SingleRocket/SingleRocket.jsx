import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { Badge } from 'react-bootstrap';
import style from './SingleRocket.module.css';
import { RESERVE_ROCKET } from '../../constants';

function SingleRocket({ rocket }) {
  const {
    name, flickrImages, id, description, reserved,
  } = rocket;

  const dispatch = useDispatch();

  const [rocketState, setRocketState] = useState(reserved);

  const reserveRocket = (id) => {
    setRocketState(!rocketState);
    dispatch({ type: RESERVE_ROCKET, id });
  };

  return (
    <div className={style.container}>
      <div className={style.imgContainer}>
        <img src={flickrImages} alt={name} />
      </div>
      <div className={style.infoContainer}>
        <h1>{name}</h1>
        <p>
          {rocketState && <Badge bg="primary">Reserved</Badge>}
          {' '}
          {description}
        </p>
        {rocketState ? <button id={id} onClick={(e) => reserveRocket(e.target.id)} className={style.cancelButton} type="button">Cancel Reservation</button> : <button data-testid="reserve-btn" id={id} onClick={(e) => reserveRocket(e.target.id)} className={style.infoButton} type="button">Reserve Rocket</button> }
      </div>
    </div>
  );
}

export default SingleRocket;

SingleRocket.defaultProps = {
  rocket: {
    id: null,
    name: '',
    type: '',
    description: '',
    flickrImages: '',
    reserved: false,
  },
};

SingleRocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    type: PropTypes.string,
    description: PropTypes.string,
    flickrImages: PropTypes.string,
    reserved: PropTypes.bool,
  }),
};
