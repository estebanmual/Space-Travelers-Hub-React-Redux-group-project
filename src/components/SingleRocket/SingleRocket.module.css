.container {
  display: flex;
  padding: 20px 0;
  column-gap: 10px;
}

.imgContainer img {
  width: 200px;
  height: 150px;
}

.infoContainer {
  width: 100%;
}

.infoContainer p {
  font-size: 2rem;
}

.infoButton {
  display: block;
  padding: 0.3em 1.2em;
  margin: 0 0.3em 0.3em 0;
  height: 35px;
  font-size: 1.2em;
  box-sizing: border-box;
  font-weight: 300;
  color: #ffffff;
  background-color: #0C78FB;
  text-align: center;
  transition: all 0.4s;
  border: 0;
  border-radius: 0.5em;
}

.infoButton:hover {
  background-color: #0a4ea1;
}
