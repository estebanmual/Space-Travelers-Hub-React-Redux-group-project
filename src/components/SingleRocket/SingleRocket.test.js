import renderer from 'react-test-renderer';
import {
  render, screen, fireEvent,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../Redux/storeSetUp';
import SingleRocket from './SingleRocket';

describe('testing Header', () => {
  it('snapshot for the header ', () => {
    const tree = renderer.create(
      <Provider store={store}><SingleRocket /></Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing Header component renders correctly', () => {
    render(<Provider store={store}><SingleRocket /></Provider>);
    expect(screen.getByText('Reserve Rocket')).toBeInTheDocument();
  });
  it('Clicking the button to reserve', () => {
    const mockhandler = jest.fn();
    const button = render(<Provider store={store}><SingleRocket /></Provider>).getByTestId('reserve-btn');

    if (fireEvent.click(button)) {
      mockhandler();
    }

    expect(mockhandler).toHaveBeenCalledTimes(1);
  });
  it('click button to leave rocket', () => {
    const rocket = {
      id: '',
      name: '',
      type: '',
      flickrImages: '',
      description: '',
      reserved: true,
    };
    const button = render(<Provider store={store}><SingleRocket rocket={rocket} /></Provider>).getByText('Cancel Reservation');
    const mockhandler = jest.fn();
    if (fireEvent.click(button)) { mockhandler(); }

    expect(mockhandler).toHaveBeenCalledTimes(1);
  });
});
