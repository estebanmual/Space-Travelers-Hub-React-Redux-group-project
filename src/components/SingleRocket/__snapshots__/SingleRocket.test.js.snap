// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header snapshot for the header  1`] = `
<div
  className="container"
>
  <div
    className="imgContainer"
  >
    <img
      alt=""
      src=""
    />
  </div>
  <div
    className="infoContainer"
  >
    <h1 />
    <p>
       
    </p>
    <button
      className="infoButton"
      data-testid="reserve-btn"
      id={null}
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
